pkg/vault/contracts/Vault.sol

```diff
@@ -1143,47 +1141,65 @@ contract Vault is IVaultMain, VaultCommon, Proxy {
             // surplus of underlying tokens.
 
             // Gets the amount of underlying to wrap in order to rebalance the buffer.
-            uint256 bufferUnderlyingSurplus = _getBufferUnderlyingSurplus(bufferBalances, wrappedToken);
+            uint256 bufferUnderlyingSurplus = bufferBalances.getBufferUnderlyingSurplus(wrappedToken);
             uint256 bufferWrappedSurplus;
 
-            if (bufferUnderlyingSurplus > 0) {
-                bufferWrappedSurplus = wrappedToken.convertToShares(bufferUnderlyingSurplus);
-            }
+            uint256 vaultUnderlyingDelta;
+            uint256 vaultWrappedDelta;
 
-            uint256 calculatedUnderlyingDelta;
-            uint256 calculatedWrappedDelta;
+            // When wrapping and in query mode, the vault needs to have enough reserves of underlying token to do the
+            // wrap operation. If the vault does not have enough tokens to do the actual wrap, use ERC4626 preview.
+            bool shouldPreview = _isQueryContext() &&
+                _reservesOf[underlyingToken] < amountInUnderlying + bufferUnderlyingSurplus;
```

Comment:
as discussed off-thread, while this extra check improves accuracy whenever possible, the worst case scenario is still the same, and it's probably better to just make the result of the query independent from the vault balances (here and in `unwrap`)

Reply:
Done, code and tests are adjusted